docs/design/deployment_design.md

```diff
@@ -0,0 +1,218 @@
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+-->
+
+This doc is to discuss targets and approaches to deploy Mizar to kubernetes (k8s) and Arktos.
+
+### Targets
+
+- Mizar deployment for both k8s and Arktos.
+- Mizar deployment in different environment including local dev environment (one VM), GCP environment (multiple VMs), AWS environment (multiple VMs).
+- Mizar deployment should support factors such as multi-apiserver, multi-etcd, multi-tenant.
+- Support deploy two version of Mizar: for developer and for end user.
+- Mizar setup script should support update Mizar to new version.
+
+### Requirements
+
+- Mizar deployment should minimize the steps needed to deploy Mizar and unify the steps for different deployment environments (GCP, AWS and .etc).
+- After deployment, user should be able to easily sanity check deployment result.
+
+### Definitions
+
+- **Mizar Phase 1**: current Mizar components are in the same cluster of k8s.
+
+  ![Simple Endpoint](png/mizar-phase-1.png)
+
+- **Mizar Phase 2**: Mizar is targeting to 2-cluster model. In this 2-cluster model, Mizar's management plane is running in a separate k8s cluster than Arkos compute cluster. An Arktos cluster will deploy the transit daemon and a Mizar proxy to communicate with the Mizar management plane.
+
+  Mizar cluster should be able to run either k8s or Arktos.
+
+  ![Simple Endpoint](png/mizar-phase-2.png)
+
+We should produce same major deploy process for Mizar phase 1 and 2, while deploying different components.
+
+### Environments
+
+There are different environments that Mizar should be supporting.
+
+- k8s Kind environment
+
+  Typically k8s cluster can be launched by Kind. It's k8s in one VM, but with multiple nodes.
+
+  Mizar phase 2 requires two clusters. For the two clusters, we shall test both clusters are k8s clusters, or one cluster is k8s and another cluster comes from arktos up. 
+
+- k8s GCP environment
+
+  kubernetes running in Google Cloud Platform.
+
+- k8s AWS environment
+
+  kubernetes running in AWS.
+
+- arktos-up environment
+
+  Arktos running in one machine in one node mode. It's typical Arktos environment for dev and test.
+
+- Arktos GCP environment - single master
+
+  Arktos running in GCP. It has one master node and multiple worker nodes.
+
+- Arktos AWS environment - single master
+
+  Arktos running in AWS. It has one master node and multiple worker nodes.
+
+- Arktos GCP environment - multi master
+
+  Arktos running in GCP. It has multiple API servers and multiple ETCD. Consider it has multiple master nodes.
+
+- Arktos GCP environment with Kubemark
+
+  Arktos running in GCP. It uses Kubemark to simulate large clusters with hundreds of nodes. It's a good environment to demonstrate Mizar performance in large clusters.
+
+### Dev vs Prod of Mizar
+
+All the time these should be two version of Mizar to be deployed. One is dev version and one is prod version.
```

Comment:
"At all times, there should be two versions of Mizar ready to be deployed."

Reply:
updated.